### PR TITLE
Improve Docker Setup

### LINKED ISSUE #17: Improve Docker Setup
**Problem:**
The Docker Setup (Issue #11) still lacks the database setup, to have the full system environment reproduced in Docker.
Furthermore the system works with verification emails and we could need a convenient way to test this in the development environment.
Lastly we need a way to provide the credentials for the database and the mailer, especially in Docker and in different environments (development and production).

**Solution:**
Create a database setup for standard installation & docker setup. 
For working with the mailer, we can use the tool [MailHog](https://github.com/mailhog/MailHog).
A good way to provide the credentials is via environment variables, because we can provide them via docker and in different environments.
These changes need to be documented in the readme file as well.

#18

### PR DESCRIPTION
**Pull request by matthib**
Close #17 

---

Improvements:

- Add database tables via migrations with npm package db-migrate
- Run migrations on docker-compose up
- Add plain sql of tables for documentation
- Provide MySQL credentials via environment variables and in Docker setup
- Add mailhog for email debugging
- Configure mailer with environemnt variables & provide mailer environment vars in Docker setup
- Document all changes & add instructions for Docker setup

#17

### CODE REVIEW

File: documentation/tables.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+CREATE TABLE IF NOT EXISTS `apikeys` (
```
Comment (matthib) [inline documentation/tables.sql]: @mimicc83 Do you have a SQL dump for the original database? Please correct me, if the definitions are not fully correct so we can change them.

File: docker-compose.yml [hunk 2]
```diff
@@ -17,10 +26,15 @@ services:
...
+      - MYSQL_PASS=123456
+      - MAIL_HOST=mailhog
+      - MAIL_PORT=1025
+      - MAIL_USER=mailer
+      - MAIL_PASS=123456
     depends_on:
       - db
-    command: "npm run start"
+      - mailhog
+    command: bash -c "sleep 5 && db-migrate up && npm run start"
```
Comment (roschaefer) [inline docker-compose.yml]:
```suggestion
    command: bash -c "wait-on tcp://db:3306 && db-migrate up && npm run start"
```
@matthib I don't think `sleep` is a great solution. I had multiple timing issues. Use https://www.npmjs.com/package/wait-on.

**Comment by roschaefer** [comment]
@matthib I updated the description of the PR. Once we merge this PR, issue #17 will be auto-closed. Check out [this special feature of Github](https://help.github.com/articles/closing-issues-using-keywords/).

**Comment by matthib** [comment]
> @matthib OK, good, we can merge this once the few issues are resolved.
> 
> There are no apikeys in my database so I will open another ticket for the seeds.

Regarding the Api keys: I didn't add them because of issue https://github.com/Human-Connection/Clock-of-Change-API/issues/13
You can add them manually, as the MySQL service is exposed to the host and you can connect to it via localhost (add least in Docker).

File: docker-compose.yml [hunk 3]
```diff
@@ -5,7 +5,16 @@ services:
     image: mysql:5
     environment:
       MYSQL_ROOT_PASSWORD: 123456
-      MYSQL_DATABASE: db1
+      MYSQL_DATABASE: clock_of_change
+    ports:
+      - 3306:3306
+
+  mailhog:
+    container_name: mailhog
```
[the remark this answers is not in the document]
Comment (matthib) [reply docker-compose.yml]: @roschaefer What would speak against naming the container here?

File: docker-compose.yml (hunk 2 above)
Comment (matthib) [reply docker-compose.yml]: @roschaefer See my last commit: https://github.com/Human-Connection/Clock-of-Change-API/pull/18/commits/6db1cc7305f6e8ad0fb21d6a01358646e9546f15

File: docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (roschaefer) [reply docker-compose.yml]: Redundancy. I would not give it a name except for a reason. The container name has an effect on routing etc.

**Comment by roschaefer** [comment]
@matthib if you want me to create the api keys manually, please update the README. I'm playing the ~ignorant~ unaware and will not do anything unless specified in the README.

(Well, of course I checked out the code and hoped to see some seeds but couldn't. So I tested the code up to this point)

File: docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (roschaefer) [reply docker-compose.yml]: Well, up to you:
```
robert@e480 ~> docker ps
CONTAINER ID        IMAGE                 COMMAND                  CREATED             STATUS              PORTS                                            NAMES
56983f119232        clock-of-change-api   "bash -c 'wait-on tc…"   9 minutes ago       Up 37 seconds       0.0.0.0:1337->1337/tcp                           clock-of-change-api_coc-api_1
f65fe2cadc8e        mysql:5               "docker-entrypoint.s…"   8 hours ago         Up 37 seconds       0.0.0.0:3306->3306/tcp, 33060/tcp                clock-of-change-api_db_1
2e0fa6abe643        mailhog/mailhog       "MailHog"                8 hours ago         Up 37 seconds       0.0.0.0:1025->1025/tcp, 0.0.0.0:8025->8025/tcp   mailhog
```
You can see the container name of the services are generated based on the name of the `docker-compose.yml`. If you want to give it a name, do so, I'll hit merge. :+1:

**Comment by matthib** [comment]
> @matthib if you want me to create the api keys manually, please update the README. I'm playing the ignorant unaware and will not do anything unless specified in the README.
> 
> (Well, of course I checked out the code and hoped to see some seeds but couldn't. So I tested the code up to this point)

Okay, then I'll just mention the documentation task in the relevant issue.

**Pull request merged**